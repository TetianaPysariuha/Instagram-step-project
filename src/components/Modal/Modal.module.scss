*{
    padding: 0;
    margin: 0;
}
.container{
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    z-index: 1000;
    display: flex;
    justify-content: center;
    align-items: center;
}
.containerBackground{
    background-color: rgba(94, 93, 88, 0.6);
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
}
.content{
    width: 70%;
    background-color:white;
    z-index: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    max-width: 720px;
}
.modalHeader{
    justify-self: start;
    width: 100%;
    padding: 20px 0;
    text-align: center;
    font-size: 20px;
    font-weight: 700;
    background-color: rgba($color: black, $alpha: 0.2);
    display: flex;
    align-items: center;
    justify-content: center;
    color:white;
    position: relative;
}
.text{
    padding: 20px 10px;
    color: white;
}
.btns{
    display: flex;
}
.modalHeader Button{
    position: absolute;
    right: 10px;
    top: 10px;
}

