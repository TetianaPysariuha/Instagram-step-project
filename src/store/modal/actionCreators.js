import { OPEN_MODAL, CLOSE_MODAL } from './actiions';

export const openModalAC = (payload) => ({ type: OPEN_MODAL, payload });
export const closeModalAC = () => ({ type: CLOSE_MODAL });
