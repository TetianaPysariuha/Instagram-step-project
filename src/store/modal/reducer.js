import { OPEN_MODAL, CLOSE_MODAL } from './actiions';

const initialState = {
  content: '<></>',
  isOpenModal: false,
};

// the rule is switched off because state shoul be first in props
// eslint-disable-next-line default-param-last
const modalReducer = (state = initialState, action) => {
  switch (action.type) {
    case OPEN_MODAL: {
      return { content: action.payload, isOpenModal: true };
    }
    case CLOSE_MODAL: {
      return { ...state, isOpenModal: false };
    }
    default: {
      return state;
    }
  }
};

export default modalReducer;
