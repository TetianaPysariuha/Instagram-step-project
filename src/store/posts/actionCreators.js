import {
  GET_POSTS, GET_POST_BY_ID, GET_POSTS_BY_USER_ID, ADD_POST, UPDATE_POST, DELETE_POST, SHOW_MORE_CHANGE, GET_NEW_PAGE_POSTS, CLEAR_POSTS, GET_FAVORITE_POSTS, CHANGE_ISOPENPOST,
} from './actiions';

const SERVER_URL = 'https://nameless-lake-66137.herokuapp.com';

export const getPosts = () => async (dispatch) => {
  const posts = await fetch(`${SERVER_URL}/posts`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POSTS, payload: posts });
};

export const clearPosts = () => ({ type: CLEAR_POSTS });

export const loadNewPagePosts = (payload) => async (dispatch) => {
  if (payload.start !== null && payload.end) {
    const response = await fetch(`${SERVER_URL}/posts/quantity/${payload.start}&${payload.end}`).then((res) => res.json()).then((data) => data);
    dispatch({ type: GET_NEW_PAGE_POSTS, payload: { data: response.data, countAll: response.totalCount } });
  } else {
    console.error(`For loafing posts should be start, end and userId parametres. Received: start=${payload.start},  end=${payload.end}`);
  }
};

export const getPostById = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POST_BY_ID, payload: post });
};

export const getPostsByUserId = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/user/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POSTS_BY_USER_ID, payload: post });
};

export const addNewPost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts`, {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: ADD_POST, payload: post });
};

export const updatePost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload.id}`, {
    method: 'PUT',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload.data),
  }).then((res) => res.json()).then((data) => data);
  dispatch({ type: UPDATE_POST, payload: post });
};

export const deletePost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload}`, {
    method: 'DELETE',
    mode: 'cors',
    headers: { 'Content-Type': 'aplication/json' },
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: DELETE_POST, payload: post });
};

export const showMoreChange = (payload) => ({ type: SHOW_MORE_CHANGE, payload });

export const getFavoritePostsByUserId = (payload) => async (dispatch) => {
  const posts = await fetch(`${SERVER_URL}/posts/favorite/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_FAVORITE_POSTS, payload: posts });
};

export const changeIsOpenPost = (payload) => ({ type: CHANGE_ISOPENPOST, payload });
