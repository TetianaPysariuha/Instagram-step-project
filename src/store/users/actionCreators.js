import {
  LOGGING_USER, GET_USERS, GET_USER_BY_ID, ADD_USER, UPDATE_USER, DELETE_USER, GET_SUBSCRIBERS_BY_USER_ID,
} from './actiions';

const SERVER_URL = 'https://nameless-lake-66137.herokuapp.com';

export const loggedUserReceiveData = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: LOGGING_USER, payload: user });
};

export const getUsers = () => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_USERS, payload: users });
};

export const getUserById = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_USER_BY_ID, payload: user });
};

export const addNewUser = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users`, {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: ADD_USER, payload: user });
};

export const updateUser = (payload) => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users/${payload.id}`, {
    method: 'PUT',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload.data),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: UPDATE_USER, payload: users });
};

export const deleteUser = (payload) => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users/${payload}`, {
    method: 'DELETE',
    mode: 'cors',
    headers: { 'Content-Type': 'aplication/json' },
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: DELETE_USER, payload: users });
};

export const getSubscribersByUserId = (payload) => async (dispatch) => {
  const subscribers = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_SUBSCRIBERS_BY_USER_ID, payload: subscribers });
};
