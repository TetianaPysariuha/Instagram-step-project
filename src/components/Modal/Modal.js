/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styles from './Modal.module.scss';
import { closeModalAC } from '../../store/modal/actionCreators';

function Modal() {
  const dispatch = useDispatch();
  const { content } = useSelector((store) => store.modal);

  return (
    <div className={styles.container}>
      <div data-testid="closeModal" onClick={() => { dispatch(closeModalAC()); }} className={styles.containerBackground} />
      <div className={styles.content}>
        {content}
      </div>
    </div>
  );
}

export default Modal;
