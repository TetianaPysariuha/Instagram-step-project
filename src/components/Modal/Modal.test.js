import { render, screen, fireEvent } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import Modal from './Modal';
import * as actions from '../../store/modal/actionCreators';
import '@testing-library/jest-dom';

jest.mock('react-redux');

describe('Modal snapshot testing', () => {
  test('Should Modal to render', () => {
    const content = useSelector.mockReturnValue('<p>Test Content</p>');
    const { asFragment } = render(<Modal content={content} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Modal closing work', () => {
  test('Close modal by click on background', () => {
    const content = useSelector.mockReturnValue('<p>Test Content</p>');
    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    const mockedCloseModalAC = jest.spyOn(actions, 'closeModalAC');
    render(<Modal content={content} />);
    fireEvent.click(screen.getByTestId('closeModal'));

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedCloseModalAC).toHaveBeenCalledTimes(1);
  });
});
