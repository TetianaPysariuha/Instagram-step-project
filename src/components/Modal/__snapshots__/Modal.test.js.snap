// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot testing Should Modal to render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="containerBackground"
      data-testid="closeModal"
    />
    <div
      class="content"
    />
  </div>
</DocumentFragment>
`;
